front/src/components/Main/PostList.js

```diff
@@ -1,17 +1,58 @@
-import Pagination from '@mui/material/Pagination';
 import * as React from 'react';
-import Stack from '@mui/material/Stack';
+
 import Post from './Post';
+import LoadMainPage from '../../features/LoadMainPage';
+import Previous from '../entrie/Previous';
+import Next from '../entrie/Next';
 
 export default function PostList() {
+  const [data, setData] = React.useState([]);
+  const [wholePage, setWholePage] = React.useState();
+  const [page, setPage] = React.useState(1);
+
+  console.log(data);
+
+  React.useEffect(() => {
+    LoadMainPage({current: page, no: 0})
+      .then(response => {
+        setData(response.boards);
+        setWholePage(response.wholePage);
+      })
+      .catch(error => {
+        console.error('Error:', error);
+      });
+  }, [page]);
+
+  function previous() {
+    if (page > 0) {
+      setPage(page => page - 1);
+    } else {
+      alert('첫 번째 페이지입니다.');
+    }
+  }
+
+  function next() {
+    if (page < wholePage) {
+      setPage(page => page + 1);
+    } else {
+      alert('마지막 페이지입니다.');
+    }
+  }
+
   return (
     <>
-      <Post />
-      <Post />
-      <Post />
-      <Stack spacing={4}>
-        <Pagination count={10} showFirstButton showLastButton />
-      </Stack>
+      {data.map((item, index) => (
+        <Post {...data[index]} />
+      ))}
```

Comment:
```suggestion
      {data.length !== 0 && data.map((item, index) => (
        <Post {...data[index]} />
      ))}
```
위에 해당 조건식만 하나 줘도 데이터가 있을때 `map`메소드가 실행되기 때문에 빈값일 때 에러를 해결할 수 있습니다.

Reply:
그리고 map함수를 사용할 땐, `key={}`값을 사용하는게 좋습니다. key값 안쪽에는 고유값으로 사용할 수 있는 값을 사용합니다. 저도 이유는 찾아보진 않았는데 `index`로는 `key`값을 잘 사용하지 않는다고 하더라구요